sqlalchemy_jsonapi/declarative/serializer.py

```diff
@@ -0,0 +1,118 @@
+"""A serializer for serializing SQLAlchemy models to JSON API spec."""
+
+import datetime
+
+from inflection import dasherize, underscore
+
+
+class JSONAPISerializer(object):
+    """A JSON API serializer that serializes SQLAlchemy models."""
+
+    def serialize(self, resources):
+        """Serialize resource(s) according to json-api spec."""
+        try:
+            self.fields
+            self.model
+            self.dasherize
+        except AttributeError:
+            raise
```

Comment:
I was very hesitant about this. I can do try/except in other places of the code rather than having this. 
The problem I had is if someone did not define dasherize then code later on would generate an exception if it wasn't caught. 
For example:
```
        if self.dasherize:
            mapped_fields = {x: dasherize(underscore(x)) for x in self.fields}
        else:
            mapped_fields = {x: x for x in self.fields}
```
I can put that all inside a try/except.